src/Utils/isAssignableTo.ts

```diff
@@ -182,11 +176,139 @@ export function isAssignableTo(
 
     // Check literal types
     if (source instanceof LiteralType) {
+        if (target instanceof IntrinsicType) {
```

Comment:
Ik object properties are a pain in the ass to handle and parse but most of this implementation is very specific to this use case and hard to maintain, please add more comments about why you are doing it that way.